src_review/loguru_parallel/listen.py

```diff
@@ -0,0 +1,68 @@
+import atexit
+from copy import deepcopy
+from logging.handlers import QueueListener
+from queue import Queue
+from typing import Any
+
+from loguru import logger
+from loguru._logger import Core
+
+from loguru_parallel.enqueue import create_log_queue, enqueue_logger
+
+
+def get_unconfigured_logger_instance():
+    current_core = logger._core
+    logger._core = Core()
+    new_logger = deepcopy(logger)
+    logger._core = current_core
+    new_logger.remove()
+    atexit.register(new_logger.remove)
+    return new_logger
+
+
+_unconfigured_logger = get_unconfigured_logger_instance()
+
+
+class LoguruQueueListener(QueueListener):
+    def __init__(self, handlers: list[dict[str, Any]], queue: Queue):
+        self.queue = queue
+        self._thread = None
+
+        sink_logger = deepcopy(_unconfigured_logger)
+        sink_logger.configure(handlers=handlers)
+
+        assert sink_logger is not logger
+        atexit.register(sink_logger.remove)
+        self._sink_logger = sink_logger
+
+    def handle(self, record):
+        """Logs a record from the queue."""
+        record = record.record
+        level, message = record["level"].name, record["message"]
+        self._sink_logger.patch(lambda r: r.update(record)).log(level, message)
+
+    def stop(self) -> None:
+        """Stop the listener.
+
+        This asks the thread to terminate, and then waits for it to do so.
+        Note that if you don't call this before your application exits, there
+        may be some records still left on the queue, which won't be processed.
+        """
+        if self._thread is None:
+            return
+        super().stop()
```

Comment:
Yes we could, but it causes problems if the user calls `stop` explicitly somewhere, as  I have to do in some tests for example.

Reply:
Overriding `stop` seems OK to me, because I want run `atexit.register(listener.stop)` when the listener is created and still give the user the ability to stop the listener anytime.
Btw, the `QueueListener` object is from the standard `logging` library, so it makes sense that it has to be customized a bit.